website/frontend/src/components/MetaFields/new-meta-parser.ts

```diff
@@ -0,0 +1,354 @@
+      name: data.name,
+    };
+  }
+
+  if (data.type === 'Result') {
+    return {
+      kind: data.type,
+      path: [...path, data.name],
+      value: data.value,
+      name: data.name,
+    };
+  }
+
+  if (isMetaItem(data)) {
+    if (!isField) {
+      path.push(data.name);
+    }
+    return {
+      kind: data.type,
+      path: path,
+      value: data.value,
+      name: data.name,
+    };
+  }
+
+  return null;
+}
+
+function setUpField(targetRef: MetaFieldsStruct, path: string[], key: string, select = false) {
+  setWith(targetRef, [...path, '__fields'], null, Object);
+  setWith(targetRef, [...path, '__name'], key, Object);
+  setWith(targetRef, [...path, '__path'], path.filter((item) => item !== '__fields').join('.'), Object);
+  setWith(targetRef, [...path, '__type'], '__fieldset', Object);
+  setWith(targetRef, [...path, '__select'], select, Object);
+}
+
+function processStruct(resultRef: MetaFieldsStruct, stackRef: StackItem[], current: NonNullable<StackItem>) {
+  const key = current.path.at(-1);
+  const entries = Object.entries(current.value).reverse();
+  const isSelect = entries[0].some((i) => i === '_enum' || i === '_enum_Result' || i === '_enum_Option');
+  setUpField(resultRef, current.path, key!, isSelect);
+
+  // Process fieldset field
+  if (isMetaItem(current.value)) {
+    stackRef.push(processFields(current.value, [...current.path, '__fields']));
+  } else {
+    entries.forEach(([name, value]) => {
+      if (isMetaItem(value) && isPrimitive(value)) {
+        stackRef.push(
+          processFields(
+            {
+              type: value.type,
+              name,
+              value: value,
+            },
+            [...current.path, '__fields']
+          )
+        );
+      }
+    });
+  }
+}
+
+function processStackItem(stackRef: StackItem[], payload: NonNullable<StackItem>) {
+  if (isMetaItem(payload.value)) {
+    stackRef.push(processFields(payload.value, payload.path, true));
+  } else if (Array.isArray(payload.value)) {
+    payload.value.forEach((item) => {
+      if (isMetaItem(item)) {
+        stackRef.push(processFields(item, [...payload.path, '__fields']));
+        return;
+      }
+    });
+  } else {
+    Object.entries(payload.value)
+      .reverse()
+      .forEach((item) => {
+        // fieldset
+        if (isMetaItem(item[1])) {
+          stackRef.push(processFields(item[1], [...payload.path, '__fields']));
+          return;
+        }
+      });
+  }
+}
+
+function parseField(data: MetaItem) {
+  const result: MetaFieldsStruct = {
+    __root: null,
+    __values: null,
+  };
+
+  const stack: StackItem[] = [];
+
+  stack.push(
+    processFields({
+      type: 'Struct',
+      name: '__root',
+      value: data,
+    })
+  );
+
+  while (stack.length > 0) {
+    const current = stack.pop();
+
+    if (current) {
+      if (!result.__values) {
+        result.__values = {};
+      }
+
+      // Check if this primitive
+      if (current.kind === 'Primitive') {
```

Comment:
you can make an enum describing 'kind'

Reply:
i think it is not convenient and excess to use enum in case literal types